File: examples/hello_world.cc

/* Simple hello world webserver example (C++)
   See hello_world.c for a C version */

#include <Lacewing.h>

void onGet(Lacewing::Webserver &Webserver, Lacewing::Webserver::Request &Request)
{
    Request << "Hello world!";
}

int main(int argc, char * argv[])
{
    Lacewing::EventPump EventPump;
    Lacewing::Webserver Webserver(EventPump);

    Webserver.onGet(onGet);
    Webserver.Host(80);    
    
    EventPump.StartEventLoop();
    
    return 0;
}

File: v8/shell/sample/hello_world.js

/** Load me in the shell : load('sample/hello_world.js') **/


print('Hosting hello world example...');

eventPump = new Lacewing.EventPump();
webserver = new Lacewing.Webserver(eventPump);

webserver.bind('get', function(request)
{
    print('Got request from ' + request.address + ' for "' + request.URL + '"');
    
    request.write ('Hello world from JS! <br />')
           .write ('I am ' + Lacewing.version());
});

webserver.bind('error', function(e)
{
    print('Error: ' + e);
});

webserver.host(80);
eventPump.startEventLoop();

